### ISSUE TITLE
Add Capabilities to Process and Smash Externally Supplied Data

### ISSUE THREAD

**Issue by Miserlou** [AlexsLemonade/refinebio#341]
### Context

It's quite likely that there will be researchers who will want to combine `refine.bio`-provided data with their own private data repositories.

It'd be great if we provided some functionality to make this easier for them by either running through our processors, gene-converters and smasher. Off the top of my head, I'm imagining that this could be new management command to live inside a processor/smasher Dockerfile and a small script with documentation to wire everything up together. But I'm certainly open to suggestions here!

This would also improve our own testing abilities.

### Problem or idea

I think this could be a good project for @Willv19  to take on if @kurtwheeler doesn't have a grander plan in mind.

### Solution or next step

Next step would probably to put some bounds on the requirements here - what kind of private data is a researcher most likely to have? Will it need to be processed, or just gene-conversion, or even just smashing?

**Comment by kurtwheeler** [comment]
We had a meeting to discuss the right way to implement this. The upshot of the discussion was that we should end up with a command like:
```
./external_smasher.sh -o 'GALLUS_GALLUS' -i smashed.zip -v /tmp/my_data
```
Where `-o` specifies the organism, `-i` specifies a zipfile that was downloaded from `refine.bio`, and `-v` specifies a directory which contains their data and is where the smashed data will be output.

We came up with a few ways to limit the complexity of the issue, at least initially but perhaps also longterm depending on what requests we receive:
  * We will only support a single organism
  * We will allows users to run this on raw data only.
  * We will only support Affymetrix and RNA-SEQ platforms.

This should also include a nice tutorial both explaining how to use the tool along with examples. The instructions/tutorial will need a very large and obvious warning that when users are downloading data from the UI they NEED TO SELECT NO TRANSFORMATIONS FOR THE DATA. It should explain that transforming the data before the download and after it gets resmashed with their private data will be a double transformation which will make their data wrong.

**Comment by kurtwheeler** [comment]
@jaclyn-taroni @Miserlou I believe I just wrote up everything we discussed. One thing that occurred to me as I did so was that we haven't specified a way to allow users to pick their transformation for the smashing. Should we just have a `-t` options which allows them to specify one of the valid values? I believe [these are the options](https://github.com/AlexsLemonade/refinebio/blob/dev/workers/data_refinery_workers/processors/smasher.py#L61) they'd have to pick from, am I correct? Is standard the same as picking none?

**Comment by cgreene** [comment]
Regarding
> The instructions/tutorial will need a very large and obvious warning that when users are downloading data from the UI they NEED TO SELECT NO TRANSFORMATIONS FOR THE DATA. It should explain that transforming the data before the download and after it gets resmashed with their private data will be a double transformation which will make their data wrong.

Could we add a file to the download or structured text in the readme that mentions the transformation that they used? Then we could check to make sure that the value corresponds to what we consider no transformation before smashing.

**Comment by kurtwheeler** [comment]
> Could we add a file to the download or structured text in the readme that mentions the transformation that they used? Then we could check to make sure that the value corresponds to what we consider no transformation before smashing.

Not only do I think this is a good idea to make sure that we don't end up with bad data by double-transforming, I also think this could be useful to have in case someone ever goes back to a dataset they downloaded a long time ago and forgot which transformation they used.

**Comment by jaclyn-taroni** [comment]
> Is standard the same as picking none?

No, that refers to [this transformation](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html) (z-score).

**Comment by kurtwheeler** [comment]
Ah, so then my question becomes: do we currently support applying no transformations to data?

**Comment by jaclyn-taroni** [comment]
> Should we just have a `-t` option which allows them to specify one of the valid values? 

Sounds good to me!

**Comment by Willv19** [comment]
My work towards a script is in the `Willv19-external-smasher` branch.
